File: Practica/src/desafio10/AdivinaElNumero.java
package desafio10;

import javax.swing.JOptionPane;

public class AdivinaElNumero {

  static int numeroUsuario = 0;
  static int cantidadIntentos = 0;
  static int numeroOculto = (int) (Math.random() * 100);

  public static void main(String[] args) {

    JOptionPane.showMessageDialog(null, "Existe un número oculto entre 1 y 100. Adivínalo!");
    jugarAdivinarNumero();

  }

  public static void jugarAdivinarNumero() {
    do {
      numeroUsuario = Integer.parseInt(JOptionPane.showInputDialog("Ingresa un entero entre 1 y 100"));

      if (numeroOculto > numeroUsuario) {
        JOptionPane.showMessageDialog(null, "El número oculto es mayor");
      } else if (numeroOculto < numeroUsuario) {
        JOptionPane.showMessageDialog(null, "El número oculto es menor");
      }
      cantidadIntentos++;

    } while (numeroOculto != numeroUsuario);

    JOptionPane.showMessageDialog(null, String.format("Correcto! Lo has logrado en %d intentos.", cantidadIntentos));
  }
}

File: Practica/src/test/py/adivina-el-numero.py
import random

print('Existe un número oculto entre 1 y 100. Puedes encontrarlo?')

numeroOculto: int = random.randint(1, 100)
numeroUsuario: int = 0
cantidadIntentos: int = 0

while numeroOculto != numeroUsuario:
    numeroUsuario = int(input('Ingrese un número entero: '))

    if numeroOculto > numeroUsuario:
        print('El número oculto es mayor')
    elif numeroOculto < numeroUsuario:
        print('El número oculto es menor')
    cantidadIntentos += 1

print(
    f'Correcto! Encontraste el número oculto en {cantidadIntentos} intentos.')
